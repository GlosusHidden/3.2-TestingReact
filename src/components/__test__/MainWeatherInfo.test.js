import React from 'react'
import renderer from 'react-test-renderer'

import { Provider } from 'react-redux'

import MainWeatherInfo from '../MainWeatherInfo'

import {myStore, mockStore} from './test_data/data'

describe('MainWeatherInfo Component Snapshot', () => {
  it('displayStyle - Favour', () => {
    const renderedValue = renderer.create(
      <Provider store={mockStore}>
        <MainWeatherInfo data={myStore.action.Paris} id={1} displayStyle='Favour'/>
      </Provider>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

    it('displayStyle - Geo', () => {
      const renderedValue = renderer.create(
        <MainWeatherInfo data={myStore.action.GeolocationWeather} id={null} displayStyle='Geo'/>
      ).toJSON();
      expect(renderedValue).toMatchSnapshot();
    });
});
