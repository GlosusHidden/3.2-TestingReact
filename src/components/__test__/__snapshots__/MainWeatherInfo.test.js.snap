// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainWeatherInfo Component Snapshot displayStyle - Favour 1`] = `
<div
  className="main_favour_weather_info"
>
  <div
    className="city"
  >
    Paris
  </div>
  <div
    className="img"
  >
    <img
      alt="weather"
      className="weather_icon"
      src="http://openweathermap.org/img/wn/10n@2x.png"
    />
  </div>
  <div
    className="temp"
  >
    5
    °C
  </div>
  <input
    alt="delete"
    className="img_button"
    name="submit"
    onClick={[Function]}
    src="img/del.png"
    type="image"
  />
</div>
`;

exports[`MainWeatherInfo Component Snapshot displayStyle - Geo 1`] = `
<div
  className="main_geo_weather_info"
>
  <div
    className="city"
  >
    Farforovaya
  </div>
  <div
    className="img"
  >
    <img
      alt="weather"
      className="weather_icon"
      src="http://openweathermap.org/img/wn/04n@2x.png"
    />
  </div>
  <div
    className="temp"
  >
    3
    °C
  </div>
</div>
`;
